dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/gdv/GdvFramework.kt

```diff
@@ -137,20 +136,24 @@ class GdvFramework : InDevelopmentPavedRoadFramework(
                 "euTaxonomieKompassAktivitaeten",
                 "umsatzInvestitionsaufwandFuerNachhaltigeAktivitaeten",
             ) {
-                label = "EU Taxonomie Kompass Aktivitäten"
-                explanation = "Welche Aktivitäten gem. dem EU Taxonomie-Kompass übt das Unternehmen aus?"
-                availableIf = DependsOnComponentValue(berichtspflichtUndEinwilligungZurVeroeffentlichung, "Yes")
-                setEuTaxonomieKompassAktivitaetenFixtureGenerator(this)
-                setEuTaxonomieKompassAktivitaetenViewConfigGenerator(this)
-                setEuTaxonomieKompassAktivitaetenUploadGenerator(this)
-                setEuTaxonomieKompassAktivitaetenDataModelGenerator(this)
+                setEuTaxonomieKompassAktivitaeten(this)
+                this.availableIf = DependsOnComponentValue(berichtsPflicht, "Yes")
             }
     }
 
     override fun getComponentGenerationUtils(): ComponentGenerationUtils {
         return GdvComponentGenerationUtils()
     }
 
+    private fun setEuTaxonomieKompassAktivitaeten(component: MultiSelectComponent) {
```

Comment:
There is no need for this parameter you can always pass `this`. However I do not understand why the private methods called inside which are only called here. They could also simply refer to `this`.